academic_observatory_workflows/workflows/openalex_telescope.py

```diff
@@ -0,0 +1,465 @@
+    def write_transfer_manifest(self):
+        """Write a transfer manifest file with filenames of files changed since the start date of this release.
+        The filename excludes the s3 bucket name (s3://openalex) and is between double quotes.
+        A separate manifest file is created for the download and transform bucket.
+
+        :return: The number of updated entities.
+        """
+        logging.info(
+            f"Writing info on updated entities from 'institution', 'concept' and 'work' to"
+            f" {self.transfer_manifest_path_download}"
+        )
+        logging.info(
+            f"Writing info on updated entities from 'author' and 'venue' to" f" {self.transfer_manifest_path_transform}"
+        )
+
+        s3client = boto3.client("s3")
+
+        updated_entities_count = 0
+        with open(self.transfer_manifest_path_download, "w") as f_download, open(
+            self.transfer_manifest_path_transform, "w"
+        ) as f_transform:
+            for entity in ["authors", "concepts", "institutions", "venues", "works"]:
+                manifest_obj = s3client.get_object(Bucket=OpenAlexTelescope.AWS_BUCKET, Key=f"data/{entity}/manifest")
+                content = manifest_obj["Body"].read()
+                entries = json.loads(content.decode())["entries"]
+
+                for entry in entries:
+                    updated_date_str = entry["url"].split("updated_date=")[1].split("/")[0]
+                    updated_date = pendulum.from_format(updated_date_str, "YYYY-MM-DD")
+                    if updated_date >= self.start_date:
+                        object_name = '"' + entry["url"].replace("s3://openalex/", "") + '"\n'
+                        if entity in ["authors", "venues"]:
+                            f_transform.write(object_name)
+                        else:
+                            f_download.write(object_name)
+                        updated_entities_count += 1
+
+        if updated_entities_count == 0:
+            raise AirflowSkipException("No updated entities to process")
+
+    def transfer(self, max_retries):
+        """Transfer files from AWS bucket to Google Cloud bucket
+
+        :param max_retries: Number of max retries to try the transfer
+        :return: None.
+        """
+        aws_access_key_id, aws_secret_access_key = get_aws_conn_info()
+        gc_project_id = Variable.get(AirflowVars.PROJECT_ID)
+
+        # TODO use transfer manifest instead when that is working
+        download_transfer = {"manifest": self.transfer_manifest_path_download, "bucket": self.download_bucket}
+        transform_transfer = {"manifest": self.transfer_manifest_path_transform, "bucket": self.transform_bucket}
+
+        total_count = 0
+        for transfer in [download_transfer, transform_transfer]:
+            success = False
+            prefixes = []
+            with open(transfer["manifest"], "r") as f:
+                for line in f:
+                    prefixes.append(line[1:-2])
```

Comment:
It would be good to have a comment with an example of what the manifest lines look like and what is being cut from each line.

Reply:
I've added an example of a line in the manifest file in the method above, and changed the code so it is clear that only quotes and a newline character are stripped.

`prefixes.append(line.strip("\n").strip('"'))`